tests/basic/tests.py

```diff
@@ -357,6 +357,24 @@ def test_hash(self):
             # hash)
             hash(Article())
 
+    def test_missing_hash_not_inherited(self):
+        class NoHash(models.Model):
+            def __eq__(self, other):
+                return super.__eq__(other)
+
+        with self.assertRaisesMessage(TypeError, "unhashable type: 'NoHash'"):
+            hash(NoHash(id=1))
+
+    def test_specified_parent_hash_inherited(self):
+        class ParentHash(models.Model):
+            def __eq__(self, other):
+                return super.__eq__(other)
+
+            __hash__ = models.Model.__hash__
+
+        hash(ParentHash(id=1))
+        self.assertIs(ParentHash.__hash__, models.Model.__hash__)
```

Comment:
I'm not sure if this assertion is necessary :thinking:.

Reply:
Yeah. This whole case isn’t really necessary but it shows the correct usage and the correct result. 

Half way through __new__ (the old way) we get a new class which has the parent’s __eq__, but which later has its own. 

So... I put it to assert that we get the reference we’re actually expecting. It looks obvious, but it’s been wrong for years. 

(The behaviour was changed in Python 2.6, I learnt...)

Reply:
This was the thread. It was quite interesting https://bugs.python.org/issue2235